### PR TITLE
SSCS-8792 Carers Allowance Steps Handling

### PR DESCRIPTION
**Pull request by danhamilton**
Covers SYA steps for Carers Allowance SSCS tickets:
8788
8789
8840
8790
8791
8792
8793

**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: steps/compliance/mrn-over-month-late/MRNOverOneMonthLate.js [hunk 1]
```diff
@@ -52,11 +51,14 @@ class MRNOverOneMonthLate extends SaveToDraftStore {
     const benefitType = get(this, 'journey.req.session.BenefitType.benefitType');
 
     const isDWPOfficeESA = () => useDWPOfficeESA.indexOf(benefitType) !== -1;
-    const allowUC = config.get('features.allowUC.enabled') === 'true';
+    const allowUC = isFeatureFlagEnabled('allowUC');
```
Comment (jackmaloney) [inline steps/compliance/mrn-over-month-late/MRNOverOneMonthLate.js:54]: As UC benefit switched on now, can we not just remove this?

File: steps/start/benefit-type/BenefitType.js [hunk 2]
```diff
@@ -51,15 +47,19 @@ class BenefitType extends SaveToDraftStore {
     }
 
     const allowedTypes = [benefitTypes.personalIndependencePayment];
-    if (allowESA) {
+    if (isFeatureFlagEnabled('allowESA')) {
       allowedTypes.push(benefitTypes.employmentAndSupportAllowance);
     }
-    if (allowUC) {
+    if (isFeatureFlagEnabled('allowUC')) {
```
Comment (jackmaloney) [inline steps/start/benefit-type/BenefitType.js:53]: As UC benefit switched on now, can we not just remove this?

File: steps/start/benefit-type/BenefitType.js [hunk 3]
```diff
@@ -51,15 +47,19 @@ class BenefitType extends SaveToDraftStore {
     }
 
     const allowedTypes = [benefitTypes.personalIndependencePayment];
-    if (allowESA) {
+    if (isFeatureFlagEnabled('allowESA')) {
```
Comment (jackmaloney) [inline steps/start/benefit-type/BenefitType.js:50]: As ESA benefit switched on now, can we not just remove this?

File: steps/start/independence/template.html [hunk 4]
```diff
@@ -25,10 +25,12 @@
             <form action="{{ path if path else url }}" method="post" class="form">
 
                 {% block fields %}
-                    {% if allowUC and benefitType == 'UC' %}
+                    {% if isBenefitEnabled('allowUC') and benefitType == 'UC' %}
```
Comment (jackmaloney) [inline steps/start/independence/template.html:28]: As UC benefit switched on now, can we not just remove this?

File: utils/stringUtils.js [hunk 5]
```diff
@@ -46,7 +67,8 @@ const getTribunalPanel = ben => {
   return {
     PIP: 'judge, doctor and disability expert',
     DLA: 'judge and a doctor',
-    ESA: 'judge and a doctor'
+    ESA: 'judge and a doctor',
+    carersAllowance: 'judge and a doctor'
```
Comment (jackmaloney) [inline utils/stringUtils.js:71]: I know not related to your PR, but what happens when benefit type is UC here?

Comment (danhamilton) [reply utils/stringUtils.js]: UPdated: I've added UC to this list and updated the template to use it accordingly so consistent with how the other benefits panel detaills are used here

File: steps/start/independence/template.html (hunk 4 above)
Comment (danhamilton) [reply steps/start/independence/template.html]: Done

File: steps/start/benefit-type/BenefitType.js (hunk 3 above)
Comment (danhamilton) [reply steps/start/benefit-type/BenefitType.js]: Done

File: steps/start/benefit-type/BenefitType.js (hunk 2 above)
Comment (danhamilton) [reply steps/start/benefit-type/BenefitType.js]: Done

File: steps/compliance/mrn-over-month-late/MRNOverOneMonthLate.js (hunk 1 above)
Comment (danhamilton) [reply steps/compliance/mrn-over-month-late/MRNOverOneMonthLate.js]: Done

**Review by jackmaloney — APPROVED** [review]

**Pull request merged**